plugins/lua/sort.lua

```diff
@@ -125,6 +125,109 @@ local function get_max_skill(unit_id, list)
     return max
 end
 
+local function melee_skill_effectiveness(unit_id, skill_list)
+    local unit = df.unit.find(unit_id)
+
+    -- Physical attributes
+    local strength = dfhack.units.getPhysicalAttrValue(unit, df.physical_attribute_type.STRENGTH)
+    local agility = dfhack.units.getPhysicalAttrValue(unit, df.physical_attribute_type.AGILITY)
+    local toughness = dfhack.units.getPhysicalAttrValue(unit, df.physical_attribute_type.TOUGHNESS)
+    local endurance = dfhack.units.getPhysicalAttrValue(unit, df.physical_attribute_type.ENDURANCE)
+    local body_size_base = unit.body.size_info.size_base
+
+    -- Mental attributes
+    local willpower = dfhack.units.getMentalAttrValue(unit, df.mental_attribute_type.WILLPOWER)
+    local spatial_sense = dfhack.units.getMentalAttrValue(unit, df.mental_attribute_type.SPATIAL_SENSE)
+    local kinesthetic_sense = dfhack.units.getMentalAttrValue(unit, df.mental_attribute_type.KINESTHETIC_SENSE)
+
+    -- Skills
+    local melee_skill = get_max_skill(unit_id, skill_list)
+    if melee_skill then
+        melee_skill = melee_skill.rating
+    else
+        melee_skill = 0
+    end
+    local melee_combat = dfhack.units.getNominalSkill(unit, df.job_skill.MELEE_COMBAT, true)
+
+    local rating = melee_skill * 27000 + melee_combat * 9000
+    + strength * 180 + body_size_base * 100 + kinesthetic_sense * 50 + endurance * 50
+    + agility * 30 + toughness * 20 + willpower * 20 + spatial_sense * 20
+    return rating
+end
+
+local function make_sort_by_melee_skill_effectiveness_desc(list)
+    return function(unit_id_1, unit_id_2)
+        if unit_id_1 == unit_id_2 then return 0 end
+        if unit_id_1 == -1 then return -1 end
+        if unit_id_2 == -1 then return 1 end
+        local rating1 = melee_skill_effectiveness(unit_id_1, list)
+        local rating2 = melee_skill_effectiveness(unit_id_2, list)
+        if rating1 == rating2 then return sort_by_name_desc(unit_id_1, unit_id_2) end
+        if rating1 ~= rating2 then return utils.compare(rating2, rating1) end
```

Comment:
```suggestion
        return utils.compare(rating2, rating1)
```

Reply:
similar for other instances below. I understand the stylistic choice, but I'd rather the form match the existing sort functions in this file.